x-pack/plugins/ml/public/components/job_selector/job_selector_table/job_selector_table.js

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+
+import React, { Fragment, useState, useEffect } from 'react';
+import { PropTypes } from 'prop-types';
+import { CustomSelectionTable } from '../custom_selection_table';
+import { getBadge } from '../job_selector';
+import { TimeRangeBar } from '../timerange_bar/';
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLoadingSpinner,
+  EuiTabbedContent,
+} from '@elastic/eui';
+
+import {
+  LEFT_ALIGNMENT,
+  CENTER_ALIGNMENT,
+  SortableProperties,
+} from '@elastic/eui/lib/services';
+import { i18n } from '@kbn/i18n';
+
+
+const JOB_FILTER_FIELDS = ['job_id', 'groups'];
+const GROUP_FILTER_FIELDS = ['id'];
+
+export function JobSelectorTable({
+  groupsList,
+  jobs,
+  onSelection,
+  selectedIds,
+  singleSelection,
+  timeseriesOnly
+}) {
+  const [sortableProperties, setSortableProperties] = useState();
+  const [currentTab, setCurrentTab] = useState('Jobs');
+
+  useEffect(() => {
+    let sortablePropertyItems = [];
+    let defaultSortProperty = 'job_id';
+
+    if (currentTab === 'Jobs' || singleSelection) {
+      sortablePropertyItems = [
+        {
+          name: 'job_id',
+          getValue: item => item.job_id.toLowerCase(),
+          isAscending: true,
+        },
+        {
+          name: 'groups',
+          getValue: item => (item.groups ? item.groups[0].toLowerCase() : ''),
+          isAscending: true,
+        }
+      ];
+    } else if (currentTab === 'Groups') {
+      defaultSortProperty = 'id';
+      sortablePropertyItems = [
+        {
+          name: 'id',
+          getValue: item => item.id.toLowerCase(),
+          isAscending: true,
+        }
+      ];
+    }
+    const sortableProps = new SortableProperties(sortablePropertyItems, defaultSortProperty);
+
+    setSortableProperties(sortableProps);
+  }, [jobs, currentTab]); // eslint-disable-line
```

Comment:
Which eslint error(s) are you suppressing here? I normally expect to see the ID of the rule that us being disabled with `eslint-disable-line`.

Reply:
This is disabling the react-hooks/exhaustive-deps rule. It's complaining that I've not added `singleSelection` as a dependency. It's unnecessary here as the only time `singleSelection` is incoming is when we've switched to the Single Metric Viewer which means this `useEffect` will be run again anyway as it's a different instance of the Job Selector component.
Here's more info on the rule and what's it's meant for: https://github.com/facebook/react/issues/14920